finagle-core/src/main/scala/com/twitter/finagle/tracing/Id.scala

```diff
@@ -47,12 +47,33 @@ object SpanId {
 
   def apply(spanId: Long): SpanId = new SpanId(spanId)
 
-  def fromString(spanId: String): Option[SpanId] =
-    try {
-      Some(SpanId(new RichU64String(spanId).toU64Long))
```

Comment:
mainly as I noticed other performance related code in this type. retaining the last chars means a substring, which means another string. you're right, it is far less code to do it that way. wdyt?